Exclude workflows for changes to Markdown files

yash-zededa opened the pull request:
### Summary

In this pull request, I have implemented the following changes:

1. **Added `paths-ignore` Configuration:**
   - Updated the workflow configuration to include `paths-ignore` for Markdown file changes. This adjustment ensures that the build and test processes are skipped when only Markdown files are modified. 

2. **Revised `get_run_id` Implementation:**
   - Refactored the `get_run_id` function to utilize the GitHub API instead of relying on the previous GitHub script. This change improves the reliability and efficiency of retrieving the run ID by leveraging GitHub's API directly.

OhmSpectator commented:
You have a very nice PR description!
Please put it into the commit message =)

yash-zededa commented:
> You have a very nice PR description! Please put it into the commit message =)

Yup, Doing that. Fixing yetus issues as well.

OhmSpectator commented:
Also, the two parts of the change desire to be separate commits.

OhmSpectator commented:
By the two commits, I meant those:

1st:
Updated the workflow configuration to include paths-ignore for Markdown
file changes. This adjustment ensures that the build and test processes
are skipped when only Markdown files are modified.

2nd:
Refactored the get_run_id function to utilize the GitHub API instead of
relying on the previous GitHub script. This change improves the
reliability and efficiency of retrieving the run ID by
leveraging GitHub's API directly.

yash-zededa commented:
> By the two commits, I meant those:
> 
> 1st: Updated the workflow configuration to include paths-ignore for Markdown file changes. This adjustment ensures that the build and test processes are skipped when only Markdown files are modified.
> 
> 2nd: Refactored the get_run_id function to utilize the GitHub API instead of relying on the previous GitHub script. This change improves the reliability and efficiency of retrieving the run ID by leveraging GitHub's API directly.

ok, both those are not 2 commits but like 2 points. I agree with the idea of representation, I will amend the commit message. 

thanks.

OhmSpectator reviewed (approved):
Approving to test the changes in the workflow

OhmSpectator commented:
```
Run echo "github context is {
/home/runner/work/_temp/7eab4ed2-b1cb-49c4-a429-cd25e80cd8f3.sh: line 236: unexpected EOF while looking for matching `''
Error: Process completed with exit code 2.
```
=(

OhmSpectator reviewed (changes requested):
The script fails now:
```
Run echo "github context is {
/home/runner/work/_temp/7eab4ed2-b1cb-49c4-a429-cd25e80cd8f3.sh: line 236: unexpected EOF while looking for matching `''
Error: Process completed with exit code 2.
```

yash-zededa commented:
> The script fails now:
> 
> ```
> Run echo "github context is {
> /home/runner/work/_temp/7eab4ed2-b1cb-49c4-a429-cd25e80cd8f3.sh: line 236: unexpected EOF while looking for matching `''
> Error: Process completed with exit code 2.
> ```

Fixed it, printing the context directly with echo causes interpolation problems. 

If the changes look good, please approve so we can proceed with testing.

OhmSpectator reviewed (approved):
Approving to test the script again.
Also, please amend the fix to the main commit. One PR should not introduce a problem and solve it with the next commit =)

OhmSpectator reviewed (changes requested):
```
  RESPONSE=$(curl -s -L "https://api.github.com/repos/lf-edge/eve/actions/workflows/build.yml/runs?head_sha=5ded92778ce7b6d331ca1e3a4830db0137902e73&status=success" \
    -H "Authorization: ***" \
    -H "Accept: application/vnd.github+json")
  # Check if RESPONSE is empty
  if [ -z "$RESPONSE" ]; then
    echo "API call returned no response."
    exit 1
  fi
  RUN_ID=$(echo $RESPONSE | jq -r '.workflow_runs[0].id')
  # Check if the RUN_ID is empty
  if [ "$RUN_ID" == "null" ] || [ -z "$RUN_ID" ]; then
    echo "No successful runs found"
    exit 1
  fi
  echo $RUN_ID
  echo "run_id=$RUN_ID" >> $GITHUB_OUTPUT
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_TOKEN: ***
No successful runs found
```

yash-zededa commented:
> ```
>   RESPONSE=$(curl -s -L "https://api.github.com/repos/lf-edge/eve/actions/workflows/build.yml/runs?head_sha=5ded92778ce7b6d331ca1e3a4830db0137902e73&status=success" \
>     -H "Authorization: ***" \
>     -H "Accept: application/vnd.github+json")
>   # Check if RESPONSE is empty
>   if [ -z "$RESPONSE" ]; then
>     echo "API call returned no response."
>     exit 1
>   fi
>   RUN_ID=$(echo $RESPONSE | jq -r '.workflow_runs[0].id')
>   # Check if the RUN_ID is empty
>   if [ "$RUN_ID" == "null" ] || [ -z "$RUN_ID" ]; then
>     echo "No successful runs found"
>     exit 1
>   fi
>   echo $RUN_ID
>   echo "run_id=$RUN_ID" >> $GITHUB_OUTPUT
>   shell: /usr/bin/bash -e {0}
>   env:
>     GITHUB_TOKEN: ***
> No successful runs found
> ```

The error is anticipated because we included:

```yaml
paths-ignore:
  - '.github/workflows/build.yml'
```

This configuration prevents the build.yml file from triggering the workflow, which is expected since we are using `pull_request_target`. However, I'll implement a conditional check to ensure that if changes involve build.yml, the eden.yml file will be ignored.

yash-zededa commented:
> Approving to test the script again. Also, please amend the fix to the main commit. One PR should not introduce a problem and solve it with the next commit =)

I've tested the workflows on the test repositories, but there are a few conditions that might have been overlooked since lf-edge/eve has a more complex setup. Once everything is confirmed to be working correctly, I'll squash the commits and include a detailed, descriptive commit message. Hope this is fine :)

OhmSpectator commented:
> However, I'll implement a conditional check to ensure that if changes involve build.yml, the eden.yml file will be ignored.

Do you have an idea how to test this PR without merging it and creating a new test PR on top?...

yash-zededa commented:
> > However, I'll implement a conditional check to ensure that if changes involve build.yml, the eden.yml file will be ignored.
> 
> Do you have an idea how to test this PR without merging it and creating a new test PR on top?...

We don't need to merge it to test it. Only approval should be fine. I have tested in on the test repo and it's working fine. 

Pushing changes in few mins.

OhmSpectator commented:
> Pushing changes in few mins.

Only if you have nothing to do)) Otherwise, it definitely waits until Monday)

yash-zededa commented:
> > Pushing changes in few mins.
> 
> Only if you have nothing to do)) Otherwise, it definitely waits until Monday)

Just want to wrap this up. We're unnecessarily triggering `eden.yml` for PR docs.

Pushed. @OhmSpectator, please approve—hopefully for the last time! 😊

OhmSpectator reviewed (approved):
Approve to test

yash-zededa commented:
> Approve to test

Thanks, Seems to be working as expected now, I have already tested the condition for code changes in the test repo.

OhmSpectator commented:
Don't forget to make your commit message the cherry on top! You don't need to mention intermediate issues and fixes.

yash-zededa commented:
> Don't forget to make your commit message the cherry on top! You don't need to mention intermediate issues and fixes.

Since it's working as expected, I’ve updated the commit message to include the relevant details and removed the intermediate fixes.

OhmSpectator commented:
@yash-zededa, don't we backport it to the stable branches?
